### PR TITLE
Ensure existing _meta information in mapping is preserved on ALTER TABLE

### PR DESCRIPTION
**Pull request by mfussenegger**
Closes https://github.com/crate/crate/issues/11216

## Checklist

 - [x] Added an entry in `CHANGES.txt` for user facing changes
 - [x] Updated documentation & `sql_features` table for user facing changes
 - [x] Touched code is covered by tests
 - [x] [CLA](https://crate.io/community/contribute/cla/) is signed
 - [x] This does not contain breaking changes, or if it does:
    - It is released within a major release
    - It is recorded in ``CHANGES.txt``
    - It was marked as deprecated in an earlier release if possible
    - You've thought about the consequences and other components are adapted
      (E.g. AdminUI)

### CODE REVIEW

File: libs/shared/src/main/java/io/crate/common/collections/Maps.java [hunk 1]
```diff
@@ -155,4 +158,37 @@ public static void mergeInto(Map<String, Object> source,
...
+     * If `map` already contains an entry that is also present in `addition`:
+     *  - It will recurse into it if it is a map
+     *  - Skip the entry otherwise
+     **/
+    public static void extendRecursive(Map<String, Object> map, Map<String, Object> additions) {
+        extendRecursive(map, additions, (oldList, newList) -> Lists2.concatUnique(oldList, newList));
+    }
+
+    @SuppressWarnings({"unchecked", "rawtypes"})
+    public static void extendRecursive(Map<String, Object> map,
```
Comment (seut) [inline libs/shared/src/main/java/io/crate/common/collections/Maps.java:174]: could we use this also as a replacement for `TransportSchemaUpdateAction.mergeIntoSource(...)`?

**Review by seut — APPROVED** [review]
Lgtm otherwise.

File: libs/shared/src/main/java/io/crate/common/collections/Maps.java (hunk 1 above)
Comment (mfussenegger) [reply libs/shared/src/main/java/io/crate/common/collections/Maps.java:174]: I thought so at first, but `TransportSchemaUpdateAction` has some validation integrated:

```
                    if (updateAllowed(key, sourceValue, updateValue)) {
                        source.put(key, updateValue);
                    } else if (!isUpdateIgnored(path) && !sourceValue.equals(updateValue)) {
                        String fqKey = String.join(".", path) + '.' + key;
                        throw new IllegalArgumentException(
                            "Can't overwrite " + fqKey + "=" + sourceValue + " with " + updateValue);
                    }
```

This is also the reason it additionally keeps track of the `path`. Didn't feel like it is worth to generalize that requirement.

Comment (seut) [reply libs/shared/src/main/java/io/crate/common/collections/Maps.java:174]: ah right 👍

**Pull request merged**